### DISCUSSION TITLE
wlan0 issue in applience-mode

### DISCUSSION THREAD

**Discussion by rehflux** [davesteele/comitup#228] [Q&A]
Hello and thanks for the great work, comitup is just what i was looking for.
Everything is running as desired, but in appliance mode wlan0/ap is restarted every 6 minutes, which leads to connection interruptions for the clients.

Here is an excerpt from the logs:
`Jul 22 15:15:58 maximegalon wpa_supplicant[430]: wlan0: interface state ENABLED->DISABLED`
`Jul 22 15:15:58 maximegalon wpa_supplicant[430]: wlan0: AP-DISABLED`
`Jul 22 15:15:58 maximegalon kernel: ieee80211 phy0: brcmf_vif_set_mgmt_ie: vndr ie set error : -52`
`Jul 22 15:15:58 maximegalon kernel: ieee80211 phy0: brcmf_vif_set_mgmt_ie: vndr ie set error : -52`
`Jul 22 15:15:58 maximegalon wpa_supplicant[430]: wlan0: CTRL-EVENT-DISCONNECTED bssid=e4:5f:01:f7:33:bd reason=3 locally_generated=1`
`Jul 22 15:15:58 maximegalon comitup[4310]: OK`
`Jul 22 15:15:58 maximegalon NetworkManager[372]: <info>  [1690031758.5411] device (wlan0): supplicant interface state: completed -> disconnected`
`Jul 22 15:15:58 maximegalon NetworkManager[372]: <info>  [1690031758.5413] device (p2p-dev-wlan0): supplicant management interface state: completed -> disconnected`
`Jul 22 15:15:58 maximegalon comitup[4312]: OK`
`Jul 22 15:15:58 maximegalon wpa_supplicant[430]: Using interface wlan0 with hwaddr e4:5f:01:f7:33:bd and ssid "BabyWifiVpn"`
`Jul 22 15:15:58 maximegalon wpa_supplicant[430]: wlan0: interface state UNINITIALIZED->ENABLED`
`Jul 22 15:15:58 maximegalon wpa_supplicant[430]: wlan0: AP-ENABLED`
`Jul 22 15:15:58 maximegalon wpa_supplicant[430]: wlan0: CTRL-EVENT-CONNECTED - Connection to e4:5f:01:f7:33:bd completed [id=0 id_str=]`
`Jul 22 15:15:58 maximegalon NetworkManager[372]: <info>  [1690031758.6442] device (wlan0): supplicant interface state: disconnected -> completed`
`Jul 22 15:15:58 maximegalon NetworkManager[372]: <info>  [1690031758.6443] device (p2p-dev-wlan0): supplicant management interface state: disconnected -> completed`

 Is this behavior intended? If not, is there a known fix?
Thank you!

**Comment by rehflux** [comment]
Update and additional infos about my problem:
Running on RPi4B with additional USB-Wifi (RTL8812BU-Chipset, driver: https://github.com/morrownr/88x2bu-20210702)

For me it seems like the "problem" is caused in states.py in this section:

```
@timeout
def connected_timeout(dummy: int) -> None:
    active_ssid: Optional[str]
    active_ssid = nm.get_active_ssid(modemgr.get_state_device("CONNECTED"))
    log.debug("connected_timeout comparing {} to {}".format(connection, active_ssid))
    if connection != active_ssid:
        log.warning("Connection lost on timeout")
        dev = modemgr.get_state_device("CONNECTED")
        set_state("CONNECTING", candidate_connections(dev))

    if modemgr.get_mode() == modemgr.MULTI_MODE:
        wpa.check_wpa(modemgr.get_ap_device().Interface)

        active_ssid = nm.get_active_ssid(modemgr.get_state_device("HOTSPOT"))
        if not active_ssid:
            log.warning("Hotspot lost on timeout")
            set_state("HOTSPOT")
```
In verbose mode comitup.log says "connected_timeout comparing SSID to SSID", where both SSID is my home-wifi-ssid. Then it seems the wpa.check_wpa restarts the AP. If i comment the line `wpa.check_wpa(modemgr.get_ap_device().Interface)` out, everything runs well.

**Comment by davesteele** [comment]
Sorry for the delay. 

This is a hack I added 4 years ago. AP mode was eventually failing, I could not reliably detect the failure, and restarting wpa_supplicant fixed the problem. The call you disabled causes wpa_supplicant to be [restarted every 5 minutes](https://github.com/davesteele/comitup/blob/1cef19ff2e50cea1f16061022ec162136167115d/comitup/wpa.py#L13). 

It's better if the application didn't have to do that. I'd be interested to know if you have any issues with the AP staying up more than a couple of days.

Note - I've noticed that I am getting kicked off of my Zero running in appliance mode, but it is less than once every 5 minutes. So it is not every time.

**Reply by rehflux** [reply]
> It's better if the application didn't have to do that. I'd be interested to know if you have any issues with the AP staying up more than a couple of days.

I didn’t restart the Pi for 3 days, I was able to connect to the AP, but the connection was very slow and seemed to have high latency. After a restart everything was okay again
I have now swapped the WLAN devices (WLAN1 = AP now; the WiFi USB stick can AP), let’s see how the problem behaves now.

> Note - I've noticed that I am getting kicked off of my Zero running in appliance mode, but it is less than once every 5 minutes. So it is not every time.

I have also noticed this behavior, but for example, SSH connections were interrupted every time.

**Reply by davesteele** [reply]
Note to future self - the problem addressed by f5b268647b2b4f7242b4ccf8423da684cf986c5a remains, at least after a couple of days.
